### PR TITLE
Gabriel/op 238 integrate UI with subgraph

### PR DESCRIPTION
**Pull request by gabrielorleans**
Screen Shot 2024-09-09 at 16 44 41

Button create event attestation now integrated with events list UI

### CODE REVIEW

File: src/app/events/page.tsx [hunk 1]
```diff
@@ -7,6 +7,46 @@ import { useChainId } from "wagmi";
...
+  const date: Date = new Date(epoch); 
+
+  // Get the day, month, and year
+  const day: number = date.getUTCDate();
+  const month: string = date.toLocaleString('default', { month: 'long', timeZone: 'UTC' });
+  const year: number = date.getUTCFullYear();
+
+  // Format the date string
+  return `${day}, ${month} of ${year}`;
+}
```
Comment (ribeirojose) [inline src/app/events/page.tsx]: prefer using a lib like https://date-fns.org/

File: src/app/events/page.tsx [hunk 2]
```diff
@@ -7,6 +7,46 @@ import { useChainId } from "wagmi";
...
+  decodedDataJson: string;
+  expirationTime: number;
+  id: string;
+  recipient: string;
+  refUID: string;
+  revocable: boolean;
+  revocationTime: number;
+}
+
+function sortByStartsAt(data: DataEntry[]): DataEntry[] {
```
Comment (ribeirojose) [inline src/app/events/page.tsx:38]: shouldn't this done at the subgraph level?

File: src/app/events/page.tsx [hunk 3]
```diff
@@ -22,15 +62,28 @@ export default function Events() {
...
 
+  // const attestationList = useMemo(
+  //   () => //const attestationsOrdered = sortByStartsAt(data.attestations) 
+  //     data?.attestations &&
+  //     data.attestations.map((data: any) => (
+  //       <AttestationItem key={data.id} data={data.decodedDataJson} />
+  //     )),
+  //   [data?.attestations]
+    
+  // );
```
Comment (ribeirojose) [inline src/app/events/page.tsx]: 👀 is this needed?

File: src/app/home/page.tsx [hunk 4]
```diff
@@ -5,31 +5,43 @@ import Image from "next/image";
...
-                <h2 className="text-xs text-gray-700 mb-4">Brief description: Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eget bibendum purus.</h2>
-                <h3 className="text-xs bg-gray-100 w-fit border-4 border-transparent rounded-lg">Status: published</h3>
-                </div>
-            </div>
-        ))
-        }
-        </div>
 
-      </main>
+              <main className="flex min-h-screen flex-col items-center justify-between p-4 sm:p-6 md:p-12 lg:p-24">
```
Comment (ribeirojose) [inline src/app/home/page.tsx]: what exactly changed in this page?

File: src/app/home/page.tsx [hunk 5]
```diff
@@ -5,31 +5,43 @@ import Image from "next/image";
...
+                    type="text"
+                  /> 
+                </form>
+                
+                <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-6 md:gap-8 lg:gap-12 mt-10 w-full max-w-6xl">
+                  {[1, 2, 3, 4, 5, 6].map((event) => (
+                    <div key={event} className="border-2 rounded-lg w-full">
+                      <div className="p-3">
+                        <h2 className="text-base font-bold">Event #{event}</h2> 
+                        <p className="text-xs">Aug. 25th of 2024</p>
```
Comment (ribeirojose) [inline src/app/home/page.tsx]: is this still relevant given that we're now fetching events from the subgraph?

File: src/app/events/page.tsx [hunk 6]
```diff
@@ -41,20 +94,56 @@ export default function Events() {
...
+          placeholder="Search" 
+          type="text"
+        /> 
+      </form>
+
+
+      <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 mt-10 w-full gap-y-12">
+        {
+        attestationList
+        }
```
Comment (ribeirojose) [inline src/app/events/page.tsx]: please format your code when submitting a PR

File: src/app/events/page.tsx [hunk 7]
```diff
@@ -22,15 +62,28 @@ export default function Events() {
...
       data?.attestations &&
-      data.attestations.map((data: any) => (
-        <AttestationItem key={data.id} data={data.decodedDataJson} />
+      sortByStartsAt(data.attestations).map((attestation) => (
+        <AttestationItem key={attestation.id} data={attestation.decodedDataJson} />
       )),
     [data?.attestations]
   );
 
+  console.log(attestationList)
```
Comment (ribeirojose) [inline src/app/events/page.tsx]: please remove this console.log statement

File: src/app/home/page.tsx
Comment (ribeirojose) [inline src/app/home/page.tsx:1]: `src/app/home/page.tsx` and `src/app/events/page.tsx` seem essentially the same to me, is that correct? Also, do you think we really need a page called `home` or could we just default to using `/` instead?

File: src/app/events/page.tsx (hunk 2 above)
Comment (luizakp) [reply src/app/events/page.tsx:38]: the startsAt is a data inside the attestation `decodedDataJson`

```
 query getAttestationsByAttester($attester: String!) {
  attestations(
    where: { attester: {equals: $attester} }
  ) {
    id 
    attester
    recipient
    refUID
    revocable
    revocationTime
    expirationTime
    data
    decodedDataJson
  }
}
```

File: src/app/home/page.tsx (hunk 4 above)
Comment (gabrielorleans) [reply src/app/home/page.tsx]: I actually did all the work on the events page and now just moved it back to home

File: src/app/home/page.tsx (hunk 5 above)
Comment (gabrielorleans) [reply src/app/home/page.tsx]: Thanks to your comment, this page doesn't exist anymore. Now the relevant code is in the home page as it should

File: src/app/events/page.tsx (hunk 1 above)
Comment (gabrielorleans) [reply src/app/events/page.tsx]: Function was rewritten now with date-fns lib

**Pull request merged**